internal/restconfig/restconfig.go

```diff
@@ -316,36 +389,34 @@ func (rcp *Producer) ClientConfig() clientcmd.ClientConfig {
 }
 
 func (rcp *Producer) CheckVersionMismatch(cmd *cobra.Command, args []string) error {
-	config, err := rcp.ForCluster()
-	exit.OnErrorWithMessage(err, "The provided kubeconfig is invalid")
+	return rcp.RunOnSelectedContext(func(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
+		crClient := clusterInfo.ClientProducer.ForGeneral()
 
-	client, err := controllerClient.New(config.Config, controllerClient.Options{})
-	exit.OnErrorWithMessage(err, "Error creating client")
+		submariner := &v1alpha1.Submariner{}
+		err := crClient.Get(context.TODO(), controllerClient.ObjectKey{
+			Namespace: constants.OperatorNamespace,
+			Name:      names.SubmarinerCrName,
+		}, submariner)
 
-	submariner := &v1alpha1.Submariner{}
-	err = client.Get(context.TODO(), controllerClient.ObjectKey{
-		Namespace: constants.OperatorNamespace,
-		Name:      names.SubmarinerCrName,
-	}, submariner)
-
-	if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
-		return nil
-	}
+		if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
+			return nil
+		}
 
-	exit.OnErrorWithMessage(err, fmt.Sprintf("Error retrieving Submariner object %s", names.SubmarinerCrName))
+		exit.OnErrorWithMessage(err, fmt.Sprintf("Error retrieving Submariner object %s", names.SubmarinerCrName))
```

Comment:
Is this left here intentionally or should it return the error?

Reply:
It’s left intentionally for now, this function is used for old flag-handling and new flag-handling.